l3kernel/l3file.dtx

```diff
@@ -722,6 +722,27 @@
 %   searches for this string using the current \TeX{} search
 %   path and the additional paths controlled by
 %   \cs{l_file_search_path_seq}.
+%   
+%   Since \TeX{} cannot remove files, only write to them, once a file
+%   has been found during a \TeX{} run, it will exist until the end of
+%   the run unless a non-\TeX{} process intervenes. Since file operations
+%   are relatively slow, \pkg{expl3} therefore internally tracks when a
+%   file is seen, and uses this information to avoid multiple filesystem
+%   checks. See \cs{file_forget:n} for how to indicate to \pkg{expl3} that
+%   a file may have been deleted \emph{during} a \TeX{} run.
```

Comment:
a \TeX{} run so that its existance or absense can be reasserted with \file_if_exist:nTF and similar commands.